libraries/AP_HAL/utility/RingBuffer.cpp

```diff
@@ -109,10 +110,7 @@ bool ByteBuffer::advance(uint32_t n)
     return true;
 }
 
-/*
-  read len bytes without advancing the read pointer
- */
-uint32_t ByteBuffer::peekbytes(uint8_t *data, uint32_t len)
+int ByteBuffer::peekiovec(ByteBuffer::IoVec iovec[2], uint32_t len)
 {
     if (len > available()) {
         len = available();
```

Comment:
Humn... but you have to think about what thread you are and what thread would update the value returned by available(). You are reading the buffer while another thread may be writing to the buffer. So, you may have 10 bytes available to read, and this value may only *increase*. So... I think in this particular case it doesn't matter much if we call available() twice or once. Am I missing anything? Could you share your stress test?